system/jlib/jlog.cpp

```diff
@@ -86,19 +86,113 @@ void LogMsgSysInfo::deserialize(MemoryBuffer & in)
 #endif
 }
 
+class LoggingFieldColumns
+{
+    const EnumMapping MsgFieldMap[15] =
+    {
+        { MSGFIELD_msgID,     "MsgID    " },
+        { MSGFIELD_audience,  "Audience " },
+        { MSGFIELD_class,     "Class    " },
+        { MSGFIELD_detail,    "Detail     " },
+        { MSGFIELD_date,      "Date       " },
+        { MSGFIELD_microTime, "Time(micro)     " },
+        { MSGFIELD_milliTime, "Time(milli)  " },
+        { MSGFIELD_time,      "Time        " },
+        { MSGFIELD_process,   "PID   " },
+        { MSGFIELD_thread,    "TID   " },
+        { MSGFIELD_session,   "SessionID           " },
+        { MSGFIELD_node,      "Node                " },
+        { MSGFIELD_job,       "JobID  " },
+        { MSGFIELD_user,      "UserID  " },
+        { MSGFIELD_component, "Compo " }
+    };
+    const unsigned sizeMsgFieldMap = arraysize(MsgFieldMap);
+public:
+    unsigned getMaxHeaderSize()
+    {
+        // Note: return length is slightly longer than necessary as only one time field is valid
+        // but the length of all time fields are added
+        unsigned size = 0;
+        for (unsigned i=0; i<sizeMsgFieldMap; ++i)
+            size += strlen(MsgFieldMap[i].str);
+        return size+2; // 2 extra characters for \r\n
+    }
+    unsigned getPositionOfField(unsigned logfields, unsigned positionoffield)
+    {
+        unsigned pos = 0;
+        for (unsigned i=0; i<sizeMsgFieldMap; ++i)
+        {
+            if (MsgFieldMap[i].val & positionoffield)
+                return pos;
+            if (MsgFieldMap[i].val & logfields)
```

Comment:
Is there an issue with _time in combination with microTime/milliTime?

Reply:
Yes.  I'll fix this.